src/app/api/search.api.ts

```diff
@@ -71,7 +76,10 @@ export class SearchApi {
                     (
                         result: ApolloQueryResult<SearchDatasetsOverviewQuery>,
                     ) => {
-                        return result.data;
+                        if (result.data) {
```

Comment:
For future: find a way to elegantly separate succesful query execution with non-optional result and error handler